src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -228,6 +232,49 @@ public bool Any<TData>(Func<T, TData, bool> predicate, TData data)
         public ImmutableArray<T> ToImmutable()
             => this.HasOne ? ImmutableArray.Create(_one) : _many;
 
+        public T[] ToArray()
+            => this.HasOne ? new[] { _one } : _many.ToArray();
+
+        public bool SequenceEqual(OneOrMany<T> other, IEqualityComparer<T>? comparer = null)
+        {
+            comparer ??= EqualityComparer<T>.Default;
+
+            if (Count != other.Count)
+            {
+                return false;
+            }
+
+            return HasOne ? comparer.Equals(_one, other[0]) :
+                   other.HasOne ? comparer.Equals(_many[0], other._one) :
+                   System.Linq.ImmutableArrayExtensions.SequenceEqual(_many, other._many, comparer);
```

Comment:
Here we are checking `HasOne`, then we are checking `other.HasOne`.

But `HasOne` and `other.HasOne` should be the same (either both true, or both false), because of the `Count` check earlier.

So I think `other.HasOne` is a dead always-false condition.

Reply:
Oh, we could have `_many` with single element, and `HasOne` will be false. So the previous comment is actually wrong.